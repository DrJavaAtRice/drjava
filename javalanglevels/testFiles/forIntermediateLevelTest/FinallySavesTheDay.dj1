class FinallySavesTheDay {
  
  int myMethod(int i, int j, char c) {
    try {
      if (i+c < j) {
        throw new ClassNotFoundException("Someone had better catch me!");
      }
      try {
        if (c < j) {
          throw new CloneNotSupportedException("And someone had better catch me too!");
        }
        else throw new RuntimeException("No one needs to catch me");
      }
      catch (RuntimeException e) {
      }
      
      finally {
        return 5;
      }
    }
    
    catch (ClassNotFoundException e) {
      throw new RuntimeException("I caught your exception, but you don't have to catch me.");
//      return 7;
    }
  }
}