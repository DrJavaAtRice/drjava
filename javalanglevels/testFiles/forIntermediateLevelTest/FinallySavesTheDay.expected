// Language Level Converter line number map: dj*->java. Entries: 27
//     1->6         2->7         3->8         4->9         5->10        6->11        7->12        8->13   
//     9->14       10->15       11->16       12->17       13->18       14->19       15->20       16->21   
//    17->22       18->23       19->24       20->25       21->26       22->27       23->28       24->29   
//    25->30       26->31       27->32   
class FinallySavesTheDay {
  
  public int myMethod(final int i, final int j, final char c) {
    try {
      if (i+c < j) {
        throw new ClassNotFoundException("Someone had better catch me!");
      }
      try {
        if (c < j) {
          throw new CloneNotSupportedException("And someone had better catch me too!");
        }
        else throw new RuntimeException("No one needs to catch me");
      }
      catch (final RuntimeException e) {
      }
      
      finally {
        return 5;
      }
    }
    
    catch (final ClassNotFoundException e) {
      throw new RuntimeException("I caught your exception, but you don't have to catch me.");
//      return 7;
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public FinallySavesTheDay() {
    super();
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.String toString() {
    return getClass().getName() + "(" + 
        ")";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    else if ((o == null) || (! o.getClass().equals(getClass()))) {
      return false;
    }
    else {
      FinallySavesTheDay cast = ((FinallySavesTheDay) o);
      return true;
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int hashCode() {
    return getClass().hashCode();
  }
}
